packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpType.cs

```diff
@@ -558,8 +563,19 @@ public static CSharpType FromLiteral(CSharpType type, object literalValue)
         {
             if (type.IsFrameworkType)
             {
-                var literalType = new CSharpType(type.FrameworkType, type.IsNullable);
-                literalType._literal = literalValue;
+                var literalType = new CSharpType(type.FrameworkType, type.Arguments, type.IsNullable)
+                {
+                    _literal = literalValue
+                };
+
+                return literalType;
+            }
+            else if (type is { IsFrameworkType: false, Implementation: EnumTypeProvider enumType })
```

Comment:
currently, the underlying value type of literal types could be enums, because we are wrapping literals into enums in the emitter.

According to [this comment](https://github.com/Azure/autorest.csharp/issues/4630#issuecomment-2073487616), we should stop it and when it happens, this piece of code could be removed.